File: 이건희/level1/카드 뭉치.py
def solution(cards1, cards2, goal):
    result = []
    n = len(cards1)
    m = len(cards2)

    i = j = 0
    for word in goal:
        if i < n and word == cards1[i]:
            result.append(cards1[i])
            i += 1

        if j < m and word == cards2[j]:
            result.append(cards2[j])
            j += 1

    return 'Yes' if result == goal else 'No'

File: 이훈일/Programmers/Level1/카드 뭉치.java
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        List<String> cards1List = new ArrayList<>(Arrays.asList(cards1));
        List<String> cards2List = new ArrayList<>(Arrays.asList(cards2));
        List<String> goalList = new ArrayList<>(Arrays.asList(goal));
        
        for(String s : goal) {
            if(!(cards1List.size() == 0)) {
                if(cards1List.get(0).equals(s)){
                    cards1List.remove(0);
                    goalList.remove(s);
                } 
            } 
            if (!(cards2List.size() == 0)) {
                if(cards2List.get(0).equals(s)){
                    cards2List.remove(0);
                    goalList.remove(s);
                }
            }
        }
        return goalList.size() == 0 ? "Yes":"No";
    }
}
